paper-api/src/main/java/io/papermc/paper/structure/PlacementOptions.java

```diff
@@ -0,0 +1,256 @@
+     * ie: Water won't spread, fences won't connect to blocks.</p>
+     * <p>Will default to false if not set.</p>
+     *
+     * @param strict Whether to restrict block updates.
+     * @return This PlacementOptions
+     */
+    @Contract(value = "_ -> this", mutates = "this")
+    public PlacementOptions strict(boolean strict) {
+        this.strict = strict;
+        return this;
+    }
+
+    /**
+     * Whether the structure will place with strict block placement.
+     *
+     * @return Will place with strict block placement.
+     */
+    public boolean strict() {
+        return strict;
+    }
+
+    /**
+     * Set whether to apply waterlogging when placing a structure.
+     * <p>Will default to true if not set.</p>
+     *
+     * @param applyWaterlogging Whether to apply waterlogging.
+     * @return This PlacementOptions
+     */
+    @Contract(value = "_ -> this", mutates = "this")
+    public PlacementOptions applyWaterlogging(boolean applyWaterlogging) {
+        this.applyWaterlogging = applyWaterlogging;
+        return this;
+    }
+
+    /**
+     * Whether waterlogging will be applied when placing a structure.
+     *
+     * @return Waterlogging will apply.
+     */
+    public boolean applyWaterlogging() {
+        return applyWaterlogging;
+    }
+
+    /**
+     * Set the collection of {@link BlockTransformer BlockTransformers} to apply to the structure.
+     *
+     * @param blockTransformers Collection of BlockTransformers.
+     * @return This PlacementOptions.
+     */
+    @Contract(value = "_ -> this", mutates = "this")
+    public PlacementOptions blockTransformers(Collection<BlockTransformer> blockTransformers) {
+        this.blockTransformers = new ArrayList<>(blockTransformers);
+        return this;
+    }
+
+    /**
+     * Collection of {@link BlockTransformer BlockTransformers} to apply to the structure.
+     *
+     * @return BlockTransformers to apply.
+     */
+    public @UnmodifiableView Collection<BlockTransformer> blockTransformers() {
+        return new ArrayList<>(blockTransformers);
+    }
+
+    /**
+     * Set the collection of {@link EntityTransformer EntityTransformers} to apply to the structure.
+     * <p>Will default to an empty list if not included.</p>
+     *
+     * @param entityTransformers EntityTransformers to apply.
+     * @return This PlacementOptions.
+     */
+    @Contract(value = "_ -> this", mutates = "this")
+    public PlacementOptions entityTransformers(Collection<EntityTransformer> entityTransformers) {
+        this.entityTransformers = new ArrayList<>(entityTransformers);
+        return this;
+    }
+
+    /**
+     * Collection of {@link EntityTransformer EntityTransformers} to apply to the structure.
+     *
+     * @return EntityTransformers to apply.
+     */
+    public @UnmodifiableView Collection<EntityTransformer> entityTransformers() {
+        return new ArrayList<>(entityTransformers);
+    }
+
```

Comment:
These should be immutable views.

However.... I am just SO not sure about exposing these methods they technically can be package private). Yes there is already a method that does, but in general that api is *god awful* and it's still experimental so I am not sure if our official type should expose this yet.

Reply:
Consider the bottom half of the comment not really relevant until some further conversation.